src/stdlib_stringlist.f90

```diff
@@ -0,0 +1,996 @@
+! stdlib_stringlist.f90 --
+!     Module for storing and manipulating lists of strings
+!     The strings may have arbitrary lengths, not necessarily the same
+!
+!     Note: very preliminary
+!
+!     TODO:
+!     insert( list_end, ... ) in an empty list?
+!     concatenate two string lists
+!
+!     Not implemented yet:
+!     insert a list or an array of character strings
+!     replace a string, list or an array of character strings
+!     concatenate a list with another list or an array
+!
+!     Limited to implemented routines
+!
+module stdlib_stringlist
+    implicit none
+
+    private
+    public :: stringlist_type
+    public :: operator(//)
+    public :: operator(+)
+    public :: operator(-)
+    public :: list_end
+
+    type stringlist_index_type
+        private
+        logical :: head
+        integer :: offset
+    end type stringlist_index_type
+
+    type(stringlist_index_type), parameter :: list_head      = stringlist_index_type( .true., 1 )
+    type(stringlist_index_type), parameter :: list_end       = stringlist_index_type( .false., 0 )
+    type(stringlist_index_type), parameter :: list_after_end = stringlist_index_type( .false., 1 )
+
+    interface operator(+)
+        module procedure stringlist_index_add
+    end interface
+
+    interface operator(-)
+        module procedure stringlist_index_subtract
+    end interface
+
+    type string_type
```

Comment:
could the `string_type` present in `stdlib_string_type` be used instead of redefining a new one?